lib/MusicBrainz/Server/WebService/XMLSerializer.pm

```diff
@@ -281,19 +283,27 @@ sub _serialize_release_group
             Interview => 5
         );
 
-        my ($fallback) =
+        my ($fallback_name) =
             nsort_by { $fallback_type_order{$_} }
                 grep { exists $fallback_type_order{$_} }
                     map { $_->name }
                         $release_group->all_secondary_types;
+        my ($fallback_gid) =
+            nsort_by { $fallback_type_order{$_} }
+                grep { exists $fallback_type_order{$_} }
```

Comment:
Yeah, except in nsort_by/grep you should do `$fallback_type_order{$_->name}` since it's keyed by name.

The fallback stuff is used when the primary type is `Album`. If there's a more specific secondary type, it'll output that instead of `Album`. I'm guessing this is because we used to only have a flat list of release types, so you would just have "Compilation" instead of "Album (primary), Compilation (secondary)", so ouputting "Compilation" is more backwards-compatible.

That's only for the `type` attribute. We still output the primary/secondary types separately below, which people should ideally be using instead.